### PR TITLE
Consolidate OTLP variables to use TRACES / METRICS to describe talking…

### LINKED ISSUE #1352: Use either SPAN or TRACES in all related environment variables
Currently we are still mixing `TRACE` / `SPAN` in our environment variables, i.e. `OTEL_TRACE_EXPORTER` vs `OTEL_EXPORTER_OTLP_SPAN_ENDPOINT`. Decide which one is better and stick to that.

### LINKED ISSUE #1353: Rename OTEL_METRICS_EXPORTER to OTEL_METRIC_EXPORTER
Realized that our OTLP metrics exporter uses `_METRIC_` rather than `_METRICS_` in its environment variables, as [seen here](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/protocol/exporter.md#configuration-options).

IMHO it makes sense to rename the newly added `OTEL_METRICS_EXPORTER` to match that pattern (or as alternative use the `_METRICS_` pattern for the OTLP exporter).

cc @anuraaga

### PR DESCRIPTION
**Pull request by anuraaga**
… to TraceService / MetricsService

Figured we need a PR to try to knock these out :)

Fixes #1352
Fixes #1353

## Changes

Environment variables for configuring OTLP exporters are changed from SPAN / METRIC to TRACE / METRICS. This is to correspond to the services being connected to, TraceService and MetricsService. It seems natural, and leaves out ambiguity by just matching up with the service names. Hopefully these variables won't be used all that much anyways in favor of the polyglot variables.

### CODE REVIEW

**Review by iNikem — APPROVED** [review]

**Comment by carlosalberto** [comment]
I'm a little bit inclined to go with `SPAN` instead of `TRACE` (as, I think, all SIGs already use `SpanExporter`, just as the Specification itself does, as opposed to `TraceExporter`). Not a strong opinion, though.

**Comment by tedsuo** [comment]
We discussed in the Maintainers meeting, and chose TRACE.

**Review by reyang — APPROVED** [review]
LGTM.

**Review by carlosalberto — APPROVED** [review]

**Comment by carlosalberto** [comment]
@open-telemetry/specs-metrics-approvers Please somebody give us a blessing here, as Metrics is also updated ;)

File: specification/protocol/exporter.md [hunk 1]
```diff
@@ -8,12 +8,12 @@ The following configuration options MUST be available to configure the OTLP expo
...
-| Compression          | Compression key for supported compression types. Supported compression: `gzip`| No value              | `OTEL_EXPORTER_OTLP_COMPRESSION` `OTEL_EXPORTER_OTLP_SPAN_COMPRESSION` `OTEL_EXPORTER_OTLP_METRIC_COMPRESSION` |
-| Timeout              | Max waiting time for the backend to process each spans or metrics batch. | 10s               | `OTEL_EXPORTER_OTLP_TIMEOUT` `OTEL_EXPORTER_OTLP_SPAN_TIMEOUT` `OTEL_EXPORTER_OTLP_METRIC_TIMEOUT` |
+| Endpoint             | Target to which the exporter is going to send spans or metrics. The endpoint MUST be a valid URL with scheme (http or https) and host, and MAY contain a port and path. A scheme of https indicates a secure connection. When using `OTEL_EXPORTER_ENDPOINT` with OTLP/HTTP, exporters SHOULD follow the collector convention of appending the version and signal to the path (e.g. `v1/traces` or `v1/metrics`). The per-signal endpoint configuration options take precedence and can be used to override this behavior. See the [OTLP Specification][otlphttp-req] for more details. | `https://localhost:4317` | `OTEL_EXPORTER_OTLP_ENDPOINT` `OTEL_EXPORTER_OTLP_TRACE_ENDPOINT` `OTEL_EXPORTER_OTLP_METRICS_ENDPOINT` |
```
Comment (tigrannajaryan) [inline specification/protocol/exporter.md:11]:
```suggestion
| Endpoint             | Target to which the exporter is going to send spans or metrics. The endpoint MUST be a valid URL with scheme (http or https) and host, and MAY contain a port and path. A scheme of https indicates a secure connection. When using `OTEL_EXPORTER_ENDPOINT` with OTLP/HTTP, exporters SHOULD follow the collector convention of appending the version and signal to the path (e.g. `v1/traces` or `v1/metrics`). The per-signal endpoint configuration options take precedence and can be used to override this behavior. See the [OTLP Specification][otlphttp-req] for more details. | `https://localhost:4317` | `OTEL_EXPORTER_OTLP_ENDPOINT` `OTEL_EXPORTER_OTLP_TRACES_ENDPOINT` `OTEL_EXPORTER_OTLP_METRICS_ENDPOINT` |
```

File: specification/protocol/exporter.md [hunk 2]
```diff
@@ -8,12 +8,12 @@ The following configuration options MUST be available to configure the OTLP expo
...
+| Protocol             | The protocol used to transmit the data. One of `grpc`,`http/json`,`http/protobuf`. | `grpc`               | `OTEL_EXPORTER_OTLP_PROTOCOL` `OTEL_EXPORTER_OTLP_TRACE_PROTOCOL` `OTEL_EXPORTER_OTLP_METRICS_PROTOCOL` |
```
Comment (tigrannajaryan) [inline specification/protocol/exporter.md:12]:
```suggestion
| Protocol             | The protocol used to transmit the data. One of `grpc`,`http/json`,`http/protobuf`. | `grpc`               | `OTEL_EXPORTER_OTLP_PROTOCOL` `OTEL_EXPORTER_OTLP_TRACES_PROTOCOL` `OTEL_EXPORTER_OTLP_METRICS_PROTOCOL` |
```

File: specification/protocol/exporter.md [hunk 3]
```diff
@@ -8,12 +8,12 @@ The following configuration options MUST be available to configure the OTLP expo
...
+| Certificate File     | Path to certificate file for TLS credentials of gRPC client. Should only be used for a secure connection. | n/a               | `OTEL_EXPORTER_OTLP_CERTIFICATE` `OTEL_EXPORTER_OTLP_TRACE_CERTIFICATE` `OTEL_EXPORTER_OTLP_METRICS_CERTIFICATE` |
```
Comment (tigrannajaryan) [inline specification/protocol/exporter.md:13]:
```suggestion
| Certificate File     | Path to certificate file for TLS credentials of gRPC client. Should only be used for a secure connection. | n/a               | `OTEL_EXPORTER_OTLP_CERTIFICATE` `OTEL_EXPORTER_OTLP_TRACES_CERTIFICATE` `OTEL_EXPORTER_OTLP_METRICS_CERTIFICATE` |
```

File: specification/protocol/exporter.md [hunk 4]
```diff
@@ -8,12 +8,12 @@ The following configuration options MUST be available to configure the OTLP expo
...
+| Headers              | Key-value pairs to be used as headers associated with gRPC or HTTP requests. See [Specifying headers](./exporter.md#specifying-headers-via-environment-variables) for more details.                   | n/a               | `OTEL_EXPORTER_OTLP_HEADERS` `OTEL_EXPORTER_OTLP_TRACE_HEADERS` `OTEL_EXPORTER_OTLP_METRICS_HEADERS` |
```
Comment (tigrannajaryan) [inline specification/protocol/exporter.md:14]:
```suggestion
| Headers              | Key-value pairs to be used as headers associated with gRPC or HTTP requests. See [Specifying headers](./exporter.md#specifying-headers-via-environment-variables) for more details.                   | n/a               | `OTEL_EXPORTER_OTLP_HEADERS` `OTEL_EXPORTER_OTLP_TRACES_HEADERS` `OTEL_EXPORTER_OTLP_METRICS_HEADERS` |
```

File: specification/protocol/exporter.md [hunk 5]
```diff
@@ -8,12 +8,12 @@ The following configuration options MUST be available to configure the OTLP expo
...
+| Compression          | Compression key for supported compression types. Supported compression: `gzip`| No value              | `OTEL_EXPORTER_OTLP_COMPRESSION` `OTEL_EXPORTER_OTLP_TRACE_COMPRESSION` `OTEL_EXPORTER_OTLP_METRICS_COMPRESSION` |
```
Comment (tigrannajaryan) [inline specification/protocol/exporter.md:15]:
```suggestion
| Compression          | Compression key for supported compression types. Supported compression: `gzip`| No value              | `OTEL_EXPORTER_OTLP_COMPRESSION` `OTEL_EXPORTER_OTLP_TRACES_COMPRESSION` `OTEL_EXPORTER_OTLP_METRICS_COMPRESSION` |
```

File: specification/protocol/exporter.md [hunk 6]
```diff
@@ -8,12 +8,12 @@ The following configuration options MUST be available to configure the OTLP expo
...
+| Timeout              | Max waiting time for the backend to process each spans or metrics batch. | 10s               | `OTEL_EXPORTER_OTLP_TIMEOUT` `OTEL_EXPORTER_OTLP_TRACE_TIMEOUT` `OTEL_EXPORTER_OTLP_METRICS_TIMEOUT` |
```
Comment (tigrannajaryan) [inline specification/protocol/exporter.md:16]:
```suggestion
| Timeout              | Max waiting time for the backend to process each spans or metrics batch. | 10s               | `OTEL_EXPORTER_OTLP_TIMEOUT` `OTEL_EXPORTER_OTLP_TRACES_TIMEOUT` `OTEL_EXPORTER_OTLP_METRICS_TIMEOUT` |
```

File: specification/protocol/exporter.md [hunk 7]
```diff
@@ -34,8 +34,8 @@ Example 2
 Traces and metrics are sent to different collectors using different protocols:
 
 ```bash
-export OTEL_EXPORTER_OTLP_SPAN_ENDPOINT=http://collector:4317
-export OTEL_EXPORTER_OTLP_SPAN_PROTOCOL=grpc
+export OTEL_EXPORTER_OTLP_TRACE_ENDPOINT=http://collector:4317
+export OTEL_EXPORTER_OTLP_TRACE_PROTOCOL=grpc
```
Comment (tigrannajaryan) [inline specification/protocol/exporter.md:38]:
```suggestion
export OTEL_EXPORTER_OTLP_TRACES_ENDPOINT=http://collector:4317
export OTEL_EXPORTER_OTLP_TRACES_PROTOCOL=grpc
```

File: specification/protocol/exporter.md [hunk 8]
```diff
@@ -55,7 +55,7 @@ export OTEL_EXPORTER_OTLP_METRICS_PROTOCOL=http/json
 
 ### Specifying headers via environment variables
 
-The `OTEL_EXPORTER_OTLP_HEADERS`, `OTEL_EXPORTER_OTLP_SPAN_HEADERS`, `OTEL_EXPORTER_OTLP_METRIC_HEADERS` environment variables will contain a list of key value pairs, and these are expected to be represented in a format matching to the [W3C Correlation-Context](https://github.com/w3c/baggage/blob/master/baggage/HTTP_HEADER_FORMAT.md), except that additional semi-colon delimited metadata is not supported, i.e.: key1=value1,key2=value2. All attribute values MUST be considered strings.
+The `OTEL_EXPORTER_OTLP_HEADERS`, `OTEL_EXPORTER_OTLP_TRACE_HEADERS`, `OTEL_EXPORTER_OTLP_METRICS_HEADERS` environment variables will contain a list of key value pairs, and these are expected to be represented in a format matching to the [W3C Correlation-Context](https://github.com/w3c/baggage/blob/master/baggage/HTTP_HEADER_FORMAT.md), except that additional semi-colon delimited metadata is not supported, i.e.: key1=value1,key2=value2. All attribute values MUST be considered strings.
```
Comment (tigrannajaryan) [inline specification/protocol/exporter.md:58]:
```suggestion
The `OTEL_EXPORTER_OTLP_HEADERS`, `OTEL_EXPORTER_OTLP_TRACES_HEADERS`, `OTEL_EXPORTER_OTLP_METRICS_HEADERS` environment variables will contain a list of key value pairs, and these are expected to be represented in a format matching to the [W3C Correlation-Context](https://github.com/w3c/baggage/blob/master/baggage/HTTP_HEADER_FORMAT.md), except that additional semi-colon delimited metadata is not supported, i.e.: key1=value1,key2=value2. All attribute values MUST be considered strings.
```

**Review by tigrannajaryan — COMMENTED** [review]
OTLP uses plural traces, metrics and logs for consistency:
https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/protocol/otlp.md

**Comment by anuraaga** [comment]
> OTLP uses plural traces, metrics and logs for consistency:

@tigrannajaryan Really? I only see `trace` in the package, file, and service name so think it is closer

https://github.com/open-telemetry/opentelemetry-proto/tree/master/opentelemetry/proto

**Comment by tigrannajaryan** [comment]
> > OTLP uses plural traces, metrics and logs for consistency:
> 
> @tigrannajaryan Really? I only see `trace` in the package, file, and service name so think it is closer
> 
> https://github.com/open-telemetry/opentelemetry-proto/tree/master/opentelemetry/proto

We cannot change those name since it breaks compatibility and they are already declared stable. 

However, as far as I remember this was discussed and a decision was made that we should use plural for all signals going forward where we can. Collector codebase uses plural traces everywhere. OTLP/HTTP uses /v1/traces as the default URL.

**Comment by carlosalberto** [comment]
After a second look, agreed that we should go for `TRACES`, as keeping all signals plural is easier to remember IMHO (besides the compatibility reasons @tigrannajaryan gave).

**Review by bogdandrutu — COMMENTED** [review]
+1 for @tigrannajaryan comments

**Review by anuraaga — COMMENTED** [review]
Updated

**Pull request merged**